skimage/feature/_canny.py

```diff
@@ -246,6 +250,14 @@ def canny(image, sigma=1., low_threshold=None, high_threshold=None, mask=None):
     c2 = magnitude[1:, :-1][pts[:-1, 1:]]
     c_minus = c2 * w + c1 * (1 - w) <= m
     local_maxima[pts] = c_plus & c_minus
+
+    #
+    #---- If quantile_threshold is set then calculate the thresholds to use
+    #
+    if quantile_threshold:
+        high_threshold = np.percentile(magnitude, 100.0 * high_threshold)
```

Comment:
we should raise a ValueError if high_threshold is greater than 1, so that it is a more meaningful error than the one returned by ``np.percentile``.